numpy/f2py/cfuncs.py

```diff
@@ -777,7 +777,7 @@ def errmess(s: str) -> None:
         *v = PyBytes_AS_STRING(obj)[0];
         return 1;
     } else if (PyUnicode_Check(obj)) {
-        PyObject* tmp = PyUnicode_AsASCIIString(obj);
+        PyObject* tmp = PyUnicode_AsUTF8String(obj);
```

Comment:
If I understand correctly, this code is to convert one-character python `str`s into fortran 8-bits characters. I don't know what fortran's model of unicode strings is, but certainly taking the first byte of the UTF8 string is going to be the wrong thing to do. I think the old behavior was correct; throw an error if a non-ascii character was parsed.

A better approach might be to deprecate this path altogether, and force the caller to pass a `bytes` object.
Either way, a test will be needed for the change here.